### ISSUE TITLE
DisallowShortOpenTagSniff does not always recognize short open tags

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1398]
When the `short_open_tags` ini setting is turned off on the system PHPCS is being run on, short open tags used within `T_INLINE_HTML` are not recognized by the sniff.

If you have no control over the servers where code is being deployed (open source projects), this can be dangerous as there might well be short open tags remaining in the code in that case and the server on which the code is deployed might have the setting turned on.

While these can't reliably be fixed, they can be warned about when found.

Test case:

```php
Some content <? echo $var; ?> Some more content

Some content <?
echo $var;
?> Some more content
```

Related #1234
